woocommerce/woocommerce
# Avoid using DISTINCT when performing order query counts when not needed.

### Changes proposed in this Pull Request:

The use of DISTINCT when running the count query for OrderTableQuery results add a lot of overhead when the query doesn't need to be set as distinct for a proper count.

This avoids the use of `DISTINCT` when the query is solely for the order IDs and there are no joins that would potentially cause duplicates.  

So far, testing has shown that this cuts the count query time by ~15% for small stores with ~2000 orders to over 50% for stores with 500K orders.

Closes #52974.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install Debug Bar, or another plugin that allows query logging with times.
2. Install WooCommerce Subscriptions, or another extension that will likely provide more complex order queries in the admin.
3. Go to the orders admin, test searching, sorting, pagination, etc and verify that the result count and pagination still works as expected based on the number of orders that should match.
4. Review the queries with the debug bar and verify that DISTINCT is still being used if multiple columns or joins in the outer query are included.
5. Repeat the same testing in the Subscriptions admin, which will likely have more complex queries and joins.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type



-   [] Update - Update existing functionality

#### Message Improve order count query performance by avoiding usage of DISTINCT when not needed.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment Improve order count query performance by avoiding usage of DISTINCT when not needed.

</details>
